woocommerce/woocommerce-gutenberg-products-block
# Allow saved payment methods labels other than card/eCheck to display brand & last 4 digits if present

While debugging this WooCommerce Payments [issue](https://github.com/Automattic/woocommerce-payments/issues/3671), it was noticed that checkout block uses the default label `Saved token for <gateway>` if it's not a CC & eCheck.

With WooCommerce Payments, saved Credit Cards displays the label as expected. However for SEPA, it shows `Saved token for woocommerce_payments`.

This PR allows saved payment methods other than card/echeck to display brand & last 4 digits if present. E.g SEPA shows `Sepa iban ending in 3000` as expected.

### Screenshots

Screenshot before change:

![Screenshot](https://d.pr/i/vwRJ23+)

Screenshot after change:

![Screenshot](https://d.pr/i/u5FR2s+)

### Testing

#### Automated Tests

### Manual Testing

How to test the changes in this Pull Request:

* Install latest dev/release version of WooCommerce Payments & setup an account in dev mode - https://woocommerce.com/document/payments/testing/dev-mode/
* Under WooCommerce Payments -> Settings, Enable the new WooCommerce Payments checkout experience and add new payment methods. If SEPA is not on the list, let me know and I'll enable it for your test account.
  * This _should_ have enabled Euros in Multi-Currency automatically, but if not, go to WooCommerce > Settings > Multi-Currency and enable Euros as an additional currency.
* Save SEPA account to your customer account in one of two ways:
  * Add item to your cart and go to the classic checkout page, select _SEPA_, enter the account, and choose _Save payment information to my account for future purchases._, then check out.
  * Go to My Account > Payment Methods, and add the _SEPA_ method there.
  * _SEPA_ account to use`AT611904300234573201` or other test numbers available here: https://stripe.com/docs/testing#sepa-direct-debit
* Once the _SEPA_ method is saved, add an item to your cart and go to the page with the checkout block.
  * If you do not have a page with the checkout block, create a new page and add a checkout block. You may need the WooCommerce Blocks plugin installed.
  * Under _Payment options_ you will see `Saved token for woocommerce_payments` instead of `SEPA IBAN ending in 3201`
  * Checkout the branch `update/labels-for-saved-payment-methods` and rebuild the dependencies
  * Refresh the block checkout page and confirm that under _Payment options_ you will see `SEPA IBAN ending in 3201` 

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

### Changelog

> Allow saved payment methods labels other than card/eCheck to display brand & last 4 digits if present
